hfile_s3.c

```diff
@@ -1298,6 +1311,20 @@ static hFILE *s3_open_v4(const char *s3url, const char *mode, va_list *argsp) {
 
         if (fp == NULL) goto error;
 
+        if (http_response >= 300 && http_response < 400) {
```

Comment:
I think doing this for all 300 codes might be a bit over-tolerant.  It would be safer to check for the specific codes we expect for this sort of redirection.

Also, you need an `hclose_abruptly(fp);` here to avoid leakage of the old connection.

Reply:
Hi. Thank you for the review!

I specified in the latest commit the `http_response` to just check for `307`. Also, I (hopefully) fixed the leakage.